src/app/calculator/furnaces/air-heating/air-heating-form/air-heating-form.component.ts

```diff
@@ -0,0 +1,275 @@
+import { Component, ElementRef, HostListener, Input, OnInit, ViewChild } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { ModalDirective } from 'ngx-bootstrap';
+import { Subscription } from 'rxjs';
+import { PhastService } from '../../../../phast/phast.service';
+import { OperatingHours } from '../../../../shared/models/operations';
+import { AirHeatingInput } from '../../../../shared/models/phast/airHeating';
+import { FlueGasWarnings } from '../../../../shared/models/phast/losses/flueGas';
+import { Settings } from '../../../../shared/models/settings';
+import { SuiteDbService } from '../../../../suiteDb/suite-db.service';
+import { AirHeatingFormService } from '../air-heating-form.service';
+import { AirHeatingService } from '../air-heating.service';
+
+@Component({
+  selector: 'app-air-heating-form',
+  templateUrl: './air-heating-form.component.html',
+  styleUrls: ['./air-heating-form.component.css']
+})
+export class AirHeatingFormComponent implements OnInit {
+  @Input()
+  settings: Settings;
+  @Input()
+  inModal: boolean;
+
+  @ViewChild('formElement', { static: false }) formElement: ElementRef;
+  @ViewChild('gasMaterialModal', { static: false }) public gasMaterialModal: ModalDirective;
+  @ViewChild('solidMaterialModal', { static: false }) public solidMaterialModal: ModalDirective;
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    this.setOpHoursModalWidth();
+  }
+  form: FormGroup;
+  fuelOptions: any;
```

Comment:
type for fuelOptions should be : Array<FlueGasMaterial | SolidLiquidFlueGasMaterial>